core/src/main/clojure/xtdb/sql.clj

```diff
@@ -600,7 +596,7 @@
                                         ['pg_catalog table-cols]))
 
                                     (add-warning! env (->BaseTableNotFound sn tn)))
-                expr-visitor (->ExprPlanVisitor env scope)]
+                expr-visitor (->ExprPlanVisitor (assoc env :no-column-refs? true) scope)]
```

Comment:
this is what `scope`'s for - you could consider adding a scope that doesn't have any available variables